resources/tentative/charts/observable-plot.js

```diff
@@ -0,0 +1,265 @@
+import * as Plot from "@observablehq/plot";
+import { csvParse } from "d3-dsv";
+import * as d3Array from "d3-array";
+import { format as d3Format } from "d3-format";
+
+// These 2 imports use some vite machinery to directly import these files as
+// strings. Then they are constants and this reduces the load on the GC.
+import airportsString from "./datasets/airports.csv?raw";
+import flightsString from "./datasets/flights-airports.csv?raw";
+
+const ROOT = document.getElementById("chart");
+const DEFAULT_OPTIONS = {
+    // This width is really a max-width: the plot adjusts using the available
+    // width as well.
+    width: 2000,
+    // The height is especially used for the ratio.
+    height: 1000,
+};
+
+let preparedData;
+function prepare() {
+    /**
+     * AirportInformation: { state, iata, name, city, country, latitude, longitude }
+     * airports: Array<AirportInformation>
+     * flights: Array<{ origin, destination, count }>
+     */
+    const { airports, flights } = parseAirportsInformation(airportsString);
+    /**
+     * flightsByAirport: Map<iata: string, { origin: number, destination: number, total: number}>
+     */
+    const flightsByAirport = groupFlightsByAirports(flights);
+    /**
+     * byAirport: Map<iata: string, AirportInformation>
+     */
+    const byAirport = d3Array.index(airports, (d) => d.iata);
+
+    /* Array<[state, AirportInformation[]]> */
+    const airportsGroupedByStateArray = d3Array.groups(airports, (d) => d.state);
+
+    /* DescSortedArray<[{ state: string, total: number, mostUsedAirportsInState: AirportInformation[] }]> */
+    const stateInformationSortedArray = airportsGroupedByStateArray
+        .map(([state, airportsInState]) => {
+            const totalFlightsInState = d3Array.sum(airportsInState, ({ iata }) => flightsByAirport.get(iata)?.total);
+            const sorted = d3Array.sort(airportsInState, ({ iata }) => -flightsByAirport.get(iata)?.total);
+            const mostUsedAirportsInState = sorted.slice(0, 6);
```

Comment:
I'm a bit confused here. The charts in your screenshot seems to have 7 items, not 6.
That seems rather arbitrary anyway though. Why not top 10 for example instead?
Maybe this should be another configurable parameter across test runs.

Reply:
Now I'm confused too 🤔 I'll look at it closer.

Reply:
Oh yes I know: the 7th item is "Others", look at the push operation in line 71.

I picked 6 in an arbitrary way. I can extract it to a constant. What do you mean by "across test runs", do you mean we could vary it in subsequent runs? Why not...

Reply:
But TBH I don't see what varying this value would bring to the benchmark compared to just keeping it at a fixed value.

Reply:
Wouldn't varying the number force more DOM updates?

Reply:
If the initial rendering takes n milliseconds, I believe that rendering it again will take n milliseconds again, using the same operations. So a slower browser would still be slower when rendering again. And this won't test other parts of the browser engine.

But I'm in no way a browser engine expert, so I may be off here. If we do what you suggest, would that change the performance characteristics in some way (maybe running the JS code in a compiled mode?)? I'm curious now.

Reply:
I checked locally on Firefox: if I bump this value from 6 to 20, the "prepare" step doesn't change much, but "stacked chart" step increases by about 33%. But 

Would that be OK to land this as is, but that I work on a separate PR on making this value configurable, and make it vary in the steps, so that we could look at this separately?

Reply:
Landing this would also make it possible to look at the other PR more easily.

Reply:
Sounds reasonable.